ocaml/xapi/xapi_network.ml

```diff
@@ -79,39 +92,47 @@ let attach_internal ?(management_interface=false) ~__context ~self () =
   let local_pifs = Xapi_network_attach_helpers.get_local_pifs ~__context ~network:self ~host in
   let persist = try List.mem_assoc "persist" net.API.network_other_config with _ -> false in
 
-  (* Ensure internal bridge exists and is up. external bridges will be
-     	   brought up by call to interface-reconfigure. *)
-  if List.length(local_pifs) = 0 then create_internal_bridge ~__context
-      ~bridge:net.API.network_bridge ~uuid:net.API.network_uuid ~persist;
+  (* Check whether the network is managed or not. If not, the only thing that it must do is check whether the bridge exists.*)
+  if not net.API.network_managed then
+    let dbg = Context.string_of_task __context in
+    let bridges = Net.Bridge.get_all dbg () in
+    if not(List.mem net.API.network_bridge bridges) then
+       raise (Api_errors.Server_error (Api_errors.bridge_not_available, [ net.API.network_bridge ]));
+  else begin
```

Comment:
I think that this `else` belongs to the `if` on L99, which is not right. A `begin..end` from L96 to L101 would fix that.